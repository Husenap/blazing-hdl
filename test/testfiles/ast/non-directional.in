interface MyFloat {
  mant : uint
  exp : uint
  sign : bool
}

interface UnaryOpIO {
  in <- MyFloat
  out -> MyFloat
}

mod MyDuplicationUnit(io : UnaryOpIO) {
//  wire MyFloat result
//  reg MyFloat pipe
//  result.significand = io.in.significand
//  result.sign = io.in.sign
//  result.exponent = io.in.exponent + 1
//  pipe = result
//  io.out = pipe
}


