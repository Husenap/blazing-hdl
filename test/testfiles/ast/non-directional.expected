InterfaceStmt: MyFloat
  Ports: 
    PortStmt: undirected uint mant
    PortStmt: undirected uint exp
    PortStmt: undirected bool sign
InterfaceStmt: UnaryOpIO
  Ports: 
    PortStmt: input MyFloat in
    PortStmt: output MyFloat out
ModStmt: MyDuplicationUnit
  Ports: 
    PortStmt: undirected UnaryOpIO io
  Body: 
